eqt/ui/UIFormWidget.py

```diff
@@ -320,10 +345,14 @@ def saveAllWidgetStates(self):
 
     def restoreAllSavedWidgetStates(self):
         '''
-        Restore all widgets in the form to the state saved by `saveAllWidgetStates()`.
-        If `saveAllWidgetStates()` method was not previously invoked, do nothing.
+        All widgets in the form are restored to the saved states. There are saved states only if
+        `saveAllWidgetStates` was previously invoked. If there are no previously saved states,
+        `default_widget_states` are used instead, after being made visible.
         '''
-        if hasattr(self, 'widget_states'):
+        if not hasattr(self, 'widget_states'):
+            self.set_default_widget_states_visible_true()
```

Comment:
Why do we do this? What if the default state of a widget is that it is invisible and we want to re-apply that?

Reply:
Is it because the default states are added when the widgets are added which could be when the form is closed?

If so, should it be instead that the default state is saved upon first opening the form?

Reply:
Perhaps this is a later problem and we may need to open an issue about rethinking how the visibility is saved.

Reply:
I am happy with the way I saved the visibility: they are "not visible" in the default states because when we created the widgets they were, indeed, non visible. 

If I want to apply default states which are not visible to a visible form, the restoration step should inherit the visibility of the form. I tried to change the visibility after the cancel button was clicked and the form was closed hoping the widgets would inherit the visibility from the opened form but I was not successful. I do agree with you and this was not done optimally here. 

I cannot think of an example where we would want to restore non visible default states to a visible form. 

I have addressed some problems with saving default states in my next PR. Hopefully once this and the next PR are merged and we could optimise these procedures before the milestone?

Reply:
The example I was thinking of would be a case where the default case has some advanced widgets that are by default hidden and could be shown using a checkbox or similar.

But I can see the docstring explains what is happening with the visibility being restored so I am happy to approve

Reply:
Ok thank you, I will open an issue with this. #113